### PR TITLE
Update various dependencies

### PR DESCRIPTION
**Pull request by TheBigLee**
## Summary

* Update to the latest version of the service-broker
* Update Kind
* Update crossplane
* Update integration-tests
* Update kustomize

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Categorize the PR by setting a good title and adding one of the labels:
  `bug`, `enhancement`, `documentation`, `change`, `breaking`,
  as they show up in the changelog
- [ ] Update the documentation.
- [ ] Update tests.
- [ ] Link this PR to related issues.

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       matrix:
         include:
-          - kind-node-version: v1.21.2
+          - kind-node-version: v1.28.0
```
Comment (zugao) [inline .github/workflows/e2e.yml:14]: any breaking changes here? Have you checked?

Comment (TheBigLee) [reply .github/workflows/e2e.yml:14]: It's just for the e2e tests, and they still work as you can see in the GH actions

Comment (zugao) [reply .github/workflows/e2e.yml:14]: ah, I missed the file name :)

**Review by zugao — APPROVED** [review]
LGTM

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**